framework/include/utils/BSpline.h

```diff
@@ -0,0 +1,70 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Reference for B-Spline: https://mat.fsv.cvut.cz/gcg/sbornik/prochazkova.pdf
+
+#pragma once
+
+#include <vector>
+#include "libmesh/point.h"
```

Comment:
Let's put headers (here and elsewhere) into the "canonical" order?  In Foo.C, you first include Foo.h; after that (or in Foo.h) you include other MOOSE headers first, then dependency headers (like libmesh/whatever.h) second, then standard headers (like vector) third.

That makes it easier to catch bugs where a header doesn't include all it's proper dependencies but "gets away with it" because those dependencies happen to get included before it in its current use case(s).